src/fusor/fusion_matching.py

```diff
@@ -0,0 +1,266 @@
+            ):
+                match_score += 1
+            else:
+                return False, 0
+
+            if is_five_prime_partner:
+                fields_to_compare = ["exonEnd", "exonEndOffset", "elementGenomicEnd"]
+            else:
+                fields_to_compare = [
+                    "exonStart",
+                    "exonStartOffset",
+                    "elementGenomicStart",
+                ]
+
+            for field in fields_to_compare:
+                if getattr(assayed_element, field) == getattr(
+                    categorical_element, field
+                ):
+                    match_score += 1
+                else:
+                    return False, 0
+
+        return True, match_score
+
+    def _compare_fusion(
+        self, assayed_fusion: AssayedFusion, categorical_fusion: CategoricalFusion
+    ) -> bool | tuple[bool, int]:
+        """Compare assayed and categorical fusions
+        :param assayed_fusion: AssayedFusion object
+        :param categorical_fusion: CategoricalFusion object
+        :return A boolean or a tuple containing a boolean and match score
+        """
+        assayed_transcript_segments = assayed_fusion.structure
+        categorical_transcript_segments = categorical_fusion.structure
+        match_score = 0
+
+        # Check for linker elements first
+        if (
+            len(assayed_transcript_segments)
+            == len(categorical_transcript_segments)
+            == 3
+        ):
+            if assayed_transcript_segments[1] == categorical_transcript_segments[1]:
+                match_score += 1
+                assayed_transcript_segments.pop(1)
+                categorical_transcript_segments.pop(1)
+            else:
+                return False
+
+        # Compare other structural elements
+        match_data_5prime = self._compare_structure(
+            assayed_transcript_segments[0], categorical_transcript_segments[0], True
+        )
+        if not match_data_5prime[0]:
+            return False
+        match_data_3prime = self._compare_structure(
+            assayed_transcript_segments[1], categorical_transcript_segments[1], False
+        )
+        if not match_data_3prime[0]:
+            return False
+        return True, match_score + match_data_5prime[1] + match_data_3prime[1]
+
+    async def match_fusion(
+        self,
+        assayed_fusion: AssayedFusion,
+        cache_path: Path | None = None,
+    ) -> list[tuple[CategoricalFusion, int]] | None:
+        """Return best matching fusion
+        :param assayed_fusion: The assayed fusion object
+        :param cache_path: The location of the CIViC translated fusions, if present.
+            If not, the list of fusions will be generated
+        :return A list of tuples containing matching categorical fusion objects and their associated match score or None
+        """
+        fusions = []
+        categorical_fusions = (
```

Comment:
@korikuzma Same thing with this part too

Reply:
You may consider moving this work. Consider creating a `categorical_fusions` instance variable that stores this info

Reply:
Would I move this to a different module/where would the `categorical_fusions` variable be stored?